patroni/api.py

```diff
@@ -561,16 +561,28 @@ def get_postgresql_status(self, retry=False):
 
             row = self.query(stmt.format(postgresql.wal_name, postgresql.lsn_name), retry=retry)[0]
 
+            try:
+                postmaster_start_time = datetime.datetime.strptime(row[0], '%Y-%m-%d %H:%M:%S.%f %Z')
+                postmaster_start_time = postmaster_start_time.replace(tzinfo=tzutc)
+            except (TypeError, ValueError):
+                postmaster_start_time = row[0]
+
+            try:
+                replayed_timestamp = datetime.datetime.strptime(row[6], '%Y-%m-%d %H:%M:%S.%f %Z')
+                replayed_timestamp = replayed_timestamp.replace(tzinfo=tzutc)
+            except (TypeError, ValueError):
+                replayed_timestamp = row[6]
```

Comment:
This is exactly what I wanted to avoid.
`pg_catalog.pg_last_xact_replay_timestamp()` and `pg_catalog.pg_postmaster_start_time()` are returning `timestamptz`, which is converted to the python datetime object (not the `str`).

It seems that you tried to do the test while the code is in an inconsistent state. If I apply the diff and revert `78084e1617007d9223e80de4e15edbe3ffce93f7` everything works as expected.

Reply:
Ah, yes, you're right -- I installed the wheel and things are good!